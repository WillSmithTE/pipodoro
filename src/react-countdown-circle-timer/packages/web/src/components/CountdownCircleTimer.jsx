import React from 'react'
import {
  DefsLinearGradient,
  countdownCircleTimerProps,
  countdownCircleTimerDefaultProps,
} from '../../../shared'
import { useCountdown } from '../hooks'

const CountdownCircleTimer = ({
  size,
  strokeWidth,
  trailStrokeWidth,
  trailColor,
  duration,
  isPlaying,
  colors,
  strokeLinecap,
  children: Children,
  isLinearGradient,
  gradientUniqueKey,
  onComplete,
  ariaLabel,
  renderAriaTime,
  initialRemainingTime,
  rotation,
}) => {
  const {
    path,
    pathLength,
    stroke,
    strokeDashoffset,
    gradientId,
    styles,
    timeProps,
  } = useCountdown({
    isPlaying,
    size,
    // https://github.com/vydimitrov/react-countdown-circle-timer/pull/82#issuecomment-774961578
    // Find the larger strokeWidth and calculate the path.
    strokeWidth: Math.max(strokeWidth, trailStrokeWidth ?? 0),
    duration,
    initialRemainingTime,
    colors,
    isLinearGradient,
    gradientUniqueKey,
    onComplete,
    rotation,
  })

  return (
    <div style={styles.wrapperStyle} aria-label={ariaLabel}>
      <svg width={size} height={size} xmlns="http://www.w3.org/2000/svg">
        {isLinearGradient && (
          <DefsLinearGradient colors={colors} gradientId={gradientId} />
        )}
        <path
          d={path}
          fill="none"
          stroke={trailColor}
          strokeWidth={trailStrokeWidth ?? strokeWidth}
        />
        <path
          d={path}
          fill="none"
          stroke={isLinearGradient ? `url(#${gradientId})` : stroke}
          strokeLinecap={strokeLinecap}
          strokeWidth={strokeWidth}
          strokeDasharray={pathLength}
          strokeDashoffset={strokeDashoffset}
        />
      </svg>
      {Children !== null && (
        <div aria-hidden="true" style={{ ...styles.timeStyle, color: stroke }}>
          <Children {...timeProps}/>
          {/* {React.isValidElement(children)
            ? React.cloneElement(React.Children.only(children), timeProps)
            : children(timeProps)} */}
        </div>
      )}
      {typeof renderAriaTime === 'function' && (
        <div role="timer" aria-live="assertive" style={styles.visuallyHidden}>
          {renderAriaTime(timeProps)}
        </div>
      )}
    </div>
  )
}

CountdownCircleTimer.propTypes = countdownCircleTimerProps
CountdownCircleTimer.defaultProps = countdownCircleTimerDefaultProps
CountdownCircleTimer.displayName = 'CountdownCircleTimer'

export { CountdownCircleTimer }
