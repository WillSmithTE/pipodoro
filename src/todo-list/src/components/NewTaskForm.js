import { useForm } from 'react-hook-form';
export default function NewTaskForm({ newTaskHandler }) {
  const { register, handleSubmit, reset, formState, clearErrors } = useForm({
    shouldUnregister: true,
    defaultValues: { label: '' },
  });

  const onSubmit = (data) => {
    newTaskHandler(data.label);
    reset();
    clearErrors();
  };

  const errors = Object.values(formState.errors);

  return (
    <form className="new-task-form" onSubmit={handleSubmit(onSubmit)}>
      <label htmlFor="task">What's next?</label>
      <input
        id="task"
        {...register('label', {
          required: 'task cannot be blank',
          validate: {
            length: (v) =>
              v.length <= 50 || 'Task cannot be longer than 50 characters.',
          },
        })}
      />
      {errors.length > 0 && (
        <ul className="error-messages">
          {errors.map((error) => (
            <li>{error.message}</li>
          ))}
        </ul>
      )}
      <button type="submit">add</button>
    </form>
  );
}
