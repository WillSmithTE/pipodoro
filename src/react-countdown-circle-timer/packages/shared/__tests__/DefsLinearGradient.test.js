import React from 'react'
import renderer from 'react-test-renderer'

import { DefsLinearGradient } from '../components'

const fixture = {
  colors: [['#004777', 0.33], ['#F7B801', 0.33], ['#A30000']],
  gradientId: '123-abc',
}

describe('snapshot tests', () => {
  it('renders with default tags', () => {
    const tree = renderer.create(<DefsLinearGradient {...fixture} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders with custom tags', () => {
    const defs = () => null
    const linearGradient = () => null
    const stop = () => null

    const tree = renderer
      .create(
        <DefsLinearGradient
          {...fixture}
          defs={defs}
          stop={stop}
          linearGradient={linearGradient}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
