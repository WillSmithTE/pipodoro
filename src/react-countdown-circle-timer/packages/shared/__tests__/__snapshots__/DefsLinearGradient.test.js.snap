// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot tests renders with custom tags 1`] = `null`;

exports[`snapshot tests renders with default tags 1`] = `
<defs>
  <linearGradient
    id="123-abc"
    x1="1"
    x2="0"
    y1="0"
    y2="0"
  >
    <stop
      offset={0}
      stopColor="#004777"
    />
    <stop
      offset={0.33}
      stopColor="#004777"
    />
    <stop
      offset={0.66}
      stopColor="#F7B801"
    />
    <stop
      offset={1}
      stopColor="#A30000"
    />
  </linearGradient>
</defs>
`;
